enterprise/models.py

```diff
@@ -1015,14 +1030,30 @@ def __repr__(self):
         """
         return self.__str__()
 
+    @property
+    def identity_provider(self):
+        """
+        Associated identity provider instance.
+        """
+        identity_provider = utils.get_identity_provider(self.provider_id)
+        return identity_provider
+
     @property
     def provider_name(self):
         """
         Readable name for the identity provider.
         """
-        identity_provider = utils.get_identity_provider(self.provider_id)
+        identity_provider = self.identity_provider
         return identity_provider and identity_provider.name
 
+    @property
+    def sync_learner_profile_data(self):
+        """
+        Should learner profile data received from the identity provider be synced to the edX profile.
```

Comment:
sorry to nitpick again, but, should this say `Should return...`? it sounds like a question to me when I read it

Reply:
ooooh i see what it's saying. hmm

Reply:
My personal recommendation would be to replace "Should" with "Return bool indicating if" (or something like that)